Riot/Routers/NavigationRouter.swift

```diff
@@ -291,6 +293,13 @@ final class NavigationRouter: NSObject, NavigationRouterType {
         return navigationController
     }
     
+    // MARK: - Theme management
+    
+    @objc private func themeDidChange() {
+        self.navigationController.overrideUserInterfaceStyle = ThemeService.shared().isCurrentThemeDark() ? .dark : .light
+        ThemeService.shared().theme.applyStyle(onNavigationBar: self.navigationController.navigationBar, withModernScrollEdgeAppearance: false)
```

Comment:
I don't think this line is very sensible. E.g. it removes the `scrollEdgeAppearance` from some screens such as space settings and the authentication flow (if you toggle dark mode while the flow is active).